okta/resource_okta_user.go

```diff
@@ -291,7 +309,35 @@ func resourceUserCreate(d *schema.ResourceData, m interface{}) error {
 		qp = query.NewQueryParams(query.WithActivate(false))
 	}
 
-	userBody := okta.User{Profile: profile}
+	password := d.Get("password").(string)
+	recoveryQuestion := d.Get("recovery_question").(string)
+	recoveryAnswer := d.Get("recovery_answer").(string)
+
+	p := &okta.PasswordCredential{
+		Value: password,
+	}
+
+	rq := &okta.RecoveryQuestionCredential{
+		Question: recoveryQuestion,
+		Answer:   recoveryAnswer,
+	}
+
+	uc := &okta.UserCredentials{
+		Password: p,
+	}
+
+	if recoveryQuestion != "" {
+		uc := &okta.UserCredentials{
```

Comment:
Just found a bug in this code where `recovery_question` isn't being set for a user. Will fix it.
Will also need to update the tests to check if it is set. Sorry for this half-baked PR. :|

Reply:
No worries at all. Thanks for the contribution. I will wait to review it.

Reply:
@quantumew - I have fixed the bug and tested it. Also ran the user resource related acceptance tests. Review when you can.

Reply:
@quantumew - I have fixed the bug and tested it. Also ran the user resource related acceptance tests. Review when you can.